### ISSUE TITLE
Upgrade storage integration tests to Storage v2 API

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger#6366]
Currently our storage integration tests (`plugin/storage/integration/` and `cmd/jaeger/internal/integration/`) operate in the existing v1 storage APIs. We want to migrate them to use V2 api from `storage_v2/`.

The way these tests operate is each storage backend has its own entry point to the tests and that entry point is responsible for initializing various storage reader/write APIs https://github.com/jaegertracing/jaeger/blob/d6456fb13b16d4694bc3ba32fa21d7c6d302d9b2/plugin/storage/integration/integration.go#L43-L50

We can begin upgrading to v2 API by incrementally swapping these fields to have the corresponding v2 interfaces, e.g. replacing `SpanReader spanstore.Reader` field with `TraceReader tracestore.Reader` and adjusting the test functions using that interface accordingly. The backend entry points then can provide TraceReader from the existing v1 SpanReader by wrapping it in `storage_v2/v1adapter`.

Then in the following PRs we can upgrade the remaining interfaces

 * [x] SpanReader #6388
 * [x] SpanWriter #6437 
 * [x] ArchiveSpanReader #6489
 * [x] ArchiveSpanWriter #6489
 * [x] DependencyWriter #6559
 * [x] DependencyReader #6477

**Comment by mainali123** [comment]
@yurishkuro I am kinda new to opensource contribution. Can I take over this issue?

**Comment by yurishkuro** [comment]
Sure. You cannot learn if you don't try.

**Comment by Ayush-Vish** [comment]
Hi @yurishkuro  I hope you're doing well. I'm relatively new to Jaeger and I'm excited about the opportunity to contribute. I’d love to get involved and make some meaningful contributions. I am looking forward to learning from the community!

**Comment by Pushpendra24-7** [comment]
Hi @yurishkuro  can you tell how to download all the dependencies for jaeger in order to contribute in it while applying the command make build it is not working . I want to contribute in this project please guide me on that

**Comment by Pushpendra24-7** [comment]
Hi @yurishkuro their is no integration_test.go present for storage related test so should i need to make on to define storage integration test

**Comment by Pushpendra24-7** [comment]
Ok

**Comment by yurishkuro** [comment]
@ekefan yes we need to update the business logic of the tests to use ptrace model. We need to find a way to do that incrementally, not in a single humongous PR. Maybe for some time we will need to    live in a state where implementations provide both v1 and v2 storage objects while we gradually transition  the tests. 

We don't need to change json fixtures just yet, because we can easily transform them from model to ptrace.

**Comment by ekefan** [comment]
Hello @yurishkuro

I'm working on replacing `spanstore.Writer` with tracestore's writer. Before I go on, what do you think about this `model.Trace`<>`[]*model.Batches`

```go
func createBatchesFromModelTrace(jTrace model.Trace) []*model.Batch{
	spans := jTrace.Spans

	if len(spans) == 0 {
		return nil
	}
	batch := &model.Batch{}
	batch.Spans = jTrace.Spans
	return []*model.Batch{batch}
}
```

**Comment by Manik2708** [comment]
I can't see the writer in Dependency Store of storage_v2. Isn't it implemented?

**Comment by yurishkuro** [comment]
maybe not. Feel free to add it.

**Comment by Manik2708** [comment]
@yurishkuro Is dependency writer a wrapper of v1 (just like dependency reader)?

**Comment by yurishkuro** [comment]
none of v2 objects are wrappers for v1. Wrappers can exist in v1adapter.

**Comment by Manik2708** [comment]
@yurishkuro In this doc: https://docs.google.com/document/d/1s4_6VgAS7qAVp6iEm5KYvpiGw3h2Ja5T5HpKo29iv00/edit#heading=h.hlhu8pegwdcj I couldn't find anything related to writer for depstore, I could see a difference between reader of dependency store of v1 and v2 in the query parameters but would there be any difference between dependency writer of v1 and v2?

**Comment by yurishkuro** [comment]
not much of a difference logically.
